docs/good-fuzz-target.md

```diff
@@ -0,0 +1,244 @@
+# What makes a good fuzz target
+
+This document explains the concept of a *fuzz target* and
+gives advice on how to make your fuzz targets really good.
+
+# Basics
+
+A *fuzz target* is a function that accepts an array of bytes and does something interesting with these bytes using the API under test.
+We define a specific signature for fuzz targets:
+
+```cpp
+// fuzz_target.cc
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  DoSomethingInterestingWithMyAPI(Data, Size);
+  return 0;  // Non-zero return values are reserved for future use.
+}
+```
+
+Note that the interface for this function is C, and so it can be implemented in any programming language that supports C interface.
+Despite the name referencing the [LLVM libFuzzer](https://llvm.org/docs/LibFuzzer.html) fuzz targets are
+independent from a particular *fuzzing engine* and can be used with fuzzing, symbolic execution, or any other form of testing. 
+
+The basic things to remember about a fuzz target:
+
+* The fuzzing engine will execute it many times with different inputs in the same process.
```

Comment:
nit: "many times" -> "many times (ideally, thousands times per second)"

Reply:
explained in detail below.